chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/scheduler/BrokenSurveillanceRulesDAO.java

```diff
@@ -1,34 +1,99 @@
 package gov.healthit.chpl.dao.scheduler;
 
+import java.util.ArrayList;
 import java.util.List;
 
+import javax.persistence.Query;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
 import gov.healthit.chpl.dto.scheduler.BrokenSurveillanceRulesDTO;
+import gov.healthit.chpl.entity.CertificationBodyEntity;
+import gov.healthit.chpl.entity.scheduler.BrokenSurveillanceRulesEntity;
 import gov.healthit.chpl.exception.EntityCreationException;
 import gov.healthit.chpl.exception.EntityRetrievalException;
 
-/**
- * Data access object for the broken_surveillance_rules table.
- * @author alarned
- *
- */
-public interface BrokenSurveillanceRulesDAO {
-    /**
-     * Retrieves all of the broken_surveillance_rules records as a list BrokenSurveillanceRulesDTO objects.
-     * @return List<BrokenSurveillanceRulesDTO> all records
-     */
-    List<BrokenSurveillanceRulesDTO> findAll();
-
-    /**
-     * Marks all broken_surveillance_rules records as deleted.
-     */
-    void deleteAll();
-
-    /**
-     * Inserts the data in the BrokenSurveillanceRulesDTO into the broken_surveillance_rules table.
-     * @param dto BrokenSurveillanceRulesDTO object populated with the data to be inserted.
-     * @throws EntityCreationException when the data could not be inserted
-     * @throws EntityRetrievalException when the newly created record cannot be retrieved
-     */
-    void create(List<BrokenSurveillanceRulesDTO> dto)
-            throws EntityCreationException, EntityRetrievalException;
+@Repository
+public class BrokenSurveillanceRulesDAO extends BaseDAOImpl{
+
+    public List<BrokenSurveillanceRulesDTO> findAll() {
+        List<BrokenSurveillanceRulesEntity> result = this.findAllEntities();
+        List<BrokenSurveillanceRulesDTO> dtos = new ArrayList<BrokenSurveillanceRulesDTO>(result.size());
+        for (BrokenSurveillanceRulesEntity entity : result) {
+            dtos.add(new BrokenSurveillanceRulesDTO(entity));
+        }
+        return dtos;
+    }
+
+    @Transactional
+    public void deleteAll() {
+        List<BrokenSurveillanceRulesEntity> entities = this.findAllEntities();
```

Comment:
Want to make this use the same "stream()" syntax as in "InheritanceErrorsReportDAO.java"?